boards/arm/efm32hg_slstk3400a/Kconfig.board

```diff
@@ -8,4 +8,3 @@
 config BOARD_EFM32HG_SLSTK3400A
 	bool "SiLabs EFM32HG-SLSTK3400A (Happy Gecko)"
 	depends on SOC_SERIES_EFM32HG
-	select SOC_PART_NUMBER_EFM32HG322F64
```

Comment:
What's the concern w/doing a select of the SOC PART Number vs putting in defconfig?  Are there cases for a given board that we'd have different SoCs?

Reply:
I'm working with efr32_slwstk6061a board which has a replaceable module / can be used with different part numbers and somehow I assumed that other SiLabs boards are the same. They are not, in case of the remaining three the MCU is soldered on the board.

However, the principal reason I have made a change is different. After moving `SOC_PART_NUMBER_EFM32HG322F64` into a choice statement the Kconfig compilation fails if this option is selected here. 

```
Zephyr version: 1.13.99

warning: the choice symbol SOC_PART_NUMBER_EFM32HG322F64 (defined at soc/arm/silabs_exx32/efm32hg/Kconfig.soc:12) is selected by the following symbols, which has no effect: 
 - BOARD_EFM32HG_SLSTK3400A (defined at /home/piotr_m/work/zephyr/boards/arm/efm32hg_slstk3400a/Kconfig.board:8)
```
I could try to revert it back, however, I'm not clear why I see the Kconfig warning in the first place and how to get around it.

Reply:
`select` never has any effect on `choice` symbols. That warning is worded a bit confusingly now. I'll improve it when I do the next Kconfiglib update.

Maybe you could do something like this, if the available choices differ between different boards:

```
choice
	prompt "Part number"

config PART_1
	bool "Part 1"
	depends on HAS_PART_1

config PART_2
	bool "Part 2"
	depends on HAS_PART_2

config PART_3
	bool "Part 3"
	depends on HAS_PART_3

endchoice
```

The boards could then `select` the `HAS_PART_*` symbols for the parts they're compatible with.

If it's really simple, you could just put the board names directly there and do away with the `HAS_PART_*` helper symbols as well.

If you want to add defaults as well, you need to put them directly on the choice itself:

```
choice
	prompt "Part number"
	default PART_2 if BOARD_X
	default PART_3 if BOARD_Y

...

endchoice
```

See the [Kconfig tips and best practices](https://docs.zephyrproject.org/latest/application/kconfig-tips.html) for some notes re. symbol prompts. The [Setting configuration values](https://docs.zephyrproject.org/latest/porting/board_porting.html#setting-configuration-values) section in the board porting guide has some tips too.

As a side note, I can't think of a fundamental reason why `select` couldn't be supported for choice symbols, but it would complicate Kconfig semantics a lot. (Should `select` unset the other symbols? What happens if you `select` two different choice symbols? Does `imply` need to be supported now too? Does the visibility of a choice symbol now depend on other choice symbols, and not just on the `choice` itself? And more...)

Internally, the code that calculates the value of a choice symbol completely ignores `select` (but at least you get a warning, unlike in the C tools).

In practice, there's usually a cleaner solution that doesn't involve selecting visible symbols (symbols with prompts).

Reply:
Thanks for the detailed explanation! It's clear now why I see the warning. In the long run we will need to move part number selection to DTS so I would preferably not try to improve the Kconfig implementation as long as setting the option via _defconfig file is acceptable.